subprojects/docs/src/docs/userguide/directory_layout.adoc

```diff
@@ -0,0 +1,104 @@
+// limitations under the License.
+
+[[directory_layout]]
+= Directory Layout
+
+[.lead]
+Gradle uses two main directories to perform and manage its work: the <<#directory_layout:gradle_user_home_directory>> and the <<#directory_layout:project_root_directory>>.
+
+
+[[directory_layout:gradle_user_home_directory]]
+== Gradle user home directory
+
+The Gradle user home directory has roughly the following structure:
+
+[listing]
+----
+.
+├── caches // <1>
+│   ├── 4.8 // <2>
+│   ├── 4.9 // <2>
+│   ├── ⋮
+│   ├── jars-3 // <3>
+│   └── modules-2 // <3>
+├── daemon // <4>
+│   ├── ⋮
+│   ├── 4.8
+│   └── 4.9
+├── init.d // <5>
+│   └── my-setup.gradle
+├── wrapper
+│   └── dists // <6>
+│       ├── ⋮
+│       ├── gradle-4.8-bin
+│       ├── gradle-4.9-all
+│       └── gradle-4.9-bin
+└── gradle.properties // <7>
+----
+<1> Global cache directory (for everything that's not project-specific)
+<2> Version-specific caches (e.g. to support incremental builds)
+<3> Shared caches (e.g. for artifacts of dependencies)
+<4> Registry and logs of the <<gradle_daemon.adoc#gradle_daemon, Gradle Daemon>>
+<5> Global <<init_scripts.adoc#init_scripts, initialization scripts>>
+<6> Distributions downloaded by the <<gradle_wrapper.adoc#gradle_wrapper,Gradle Wrapper>>
+<7> Global <<build_environment.adoc#sec:gradle_configuration_properties,Gradle configuration properties>>
+
+[[directory_layout:gradle_user_home_directory:cache_cleanup]]
+=== Global cache cleanup
+
+Since Gradle 4.10, the Gradle user home directory is being cleaned up automatically. The cleanup is performed in the background when the Gradle daemon is stopped or shuts down. If using `--no-daemon`, it runs in the foreground after the build session with a visual progress indicator.
+
+The following cleanup strategies are applied periodically (at most every 24 hours):
+
+- Version-specific caches in `caches/<gradle-version>/` are checked for whether they are still in use. If not, directories for release versions are deleted after 30 days of inactivity, snapshot versions after 7 days of inactivity.
+- Gradle distributions in `wrapper/dists/` are checked for whether they are still in use, i.e. whether there's a corresponding version-specific cache directory. Unused distributions are deleted.
+- Shared caches in `caches/` (e.g. `jars-*`) are checked for whether they are still in use. If there's no Gradle version that still uses them, they are deleted.
+- Files in shared caches used by the current Gradle version in `caches/` (e.g. `jars-3` or `modules-2`) are checked for when they were last accessed. Depending on whether the file can be recreated locally or would have to be downloaded from a remote repository again, it will be deleted after 7 or 30 days of not being accessed, respectively.
+
+
+[[directory_layout:project_root_directory]]
+== Project root directory
+
+The project root directory has roughly the following structure:
+
+[listing,subs=+macros]
```

Comment:
❓ I'm not familiar with this. What's "subs=+macros" for?

Reply:
@eriwen Enables macro substitution in source code, specifically for `pass:quotes[_or_]` in this case.

Reply:
huh.. the more you know ☄️ 🌟